march_shared_classes/src/march_shared_classes/gait/subgait.py

```diff
@@ -19,40 +18,52 @@ def __init__(self, joints, duration, gait_type='walk_like',
         self.description = str(description)
         self.duration = duration
 
-    def to_joint_trajectory(self):
-        joint_trajectory = JointTrajectory()
-
-        timestamps = self.get_unique_timestamps()
-
-        for joint in self.joints:
-            joint_trajectory.joint_names.append(joint.name)
-
-        for timestamp in timestamps:
-            joint_trajectory_point = JointTrajectoryPoint()
-            joint_trajectory_point.time_from_start = rospy.Duration(timestamp)
-            for joint in self.joints:
-                interpolated_setpoint = joint.get_interpolated_setpoint(timestamp)
-
-                if interpolated_setpoint.time != timestamp:
-                    rospy.logerr('Time mismatch in joint ' + joint.name + ' at timestamp ' + timestamp)
-                joint_trajectory_point.positions.append(interpolated_setpoint.position)
-                joint_trajectory_point.velocities.append(interpolated_setpoint.velocity)
-            joint_trajectory.points.append(joint_trajectory_point)
-
-        return joint_trajectory
-
-    def to_setpoints(self):
-        user_defined_setpoints = []
-        timestamps = self.get_unique_timestamps()
-        for timestamp in timestamps:
-            user_defined_setpoint = Setpoint()
-            user_defined_setpoint.time_from_start = rospy.Duration.from_sec(timestamp)
-            for joint in self.joints:
-                for setpoint in joint.setpoints:
-                    if setpoint.time == timestamp:
-                        user_defined_setpoint.joint_names.append(joint.name)
-            user_defined_setpoints.append(user_defined_setpoint)
-        return user_defined_setpoints
+    @classmethod
+    def from_file(cls, robot, file_name):
+        if file_name is None or file_name == "":
+            return None
+        try:
+            gait_name = file_name.split("/")[-3]
+            subgait_name = file_name.split("/")[-2]
+            version = file_name.split("/")[-1].replace(".subgait", "")
+            subgait_dict = yaml.load(open(file_name), Loader=yaml.SafeLoader)
+            if 'gait_type' not in subgait_dict:
+                subgait_dict['gait_type'] = 'walk_like'
+        except Exception as e:
+            rospy.logerr(str(e))
```

Comment:
i would add the type of the error as well as the error itself
```suggestion
            rospy.logerr("Error occured in subgait: {}, {} ".format(type(e), e))
```
This would improve the usage of the error messages